SS-common/MPU9250/SS_MPU9250.c

```diff
@@ -155,58 +181,105 @@ MPU_STATUS SS_MPU_math_scaled_accel(MPU9250 *mpu9250) {
     return MPU_OK;
 }
 
+int32_t data[6][2000];
+int16_t  data1[6];
 MPU_STATUS SS_MPU_calibrate(MPU9250 *mpu9250) {  //Device needs to be flat during calibration!!!
-    int16_t calibration_time = 2000;
-    uint16_t max_gyro_deviation = 15;
-    uint16_t max_accel_deviation = 200;
-    HAL_Delay(100);
+
+    int16_t calibration_time = 3;                // accel sometimes return 0 than proper data that why i read only 3 times
+    uint16_t max_gyro_deviation = 50;
+    uint16_t max_accel_deviation = 150;
+    HAL_Delay(1);
     MPU_STATUS result = MPU_OK;
     uint8_t gyro_scale = mpu9250->gyro_scale;
     uint8_t accel_scale = mpu9250->accel_scale;
     uint8_t bandwidth = mpu9250->gyro_bandwidth;
     int32_t sum[] = {0, 0, 0, 0, 0, 0};  //gx, gy, gz, ax, ay, az
-    int32_t bias[6];
+    int32_t bias[5];
     result |= SS_MPU_set_gyro_scale(mpu9250, MPU_GYRO_SCALE_250);
     result |= SS_MPU_set_accel_scale(mpu9250, MPU_ACCEL_SCALE_2);
     result |= SS_MPU_set_gyro_bandwidth(mpu9250, MPU_GYRO_BAND_5);
     HAL_Delay(50);
 
-    int16_t data[6][calibration_time];
-    for(int16_t i = 0; i < calibration_time; i++) {
-        result |= SS_MPU_get_gyro_data(mpu9250);
-        result |= SS_MPU_get_accel_data(mpu9250);
-        data[0][i] = mpu9250->gyro_raw_x;
-        data[1][i] = mpu9250->gyro_raw_y;
-        data[2][i] = mpu9250->gyro_raw_z;
-        data[3][i] = mpu9250->accel_raw_x;
-        data[4][i] = mpu9250->accel_raw_y;
-        data[5][i] = mpu9250->accel_raw_z;
-        HAL_Delay(1);
-    }
-    for(uint16_t i = 0; i < calibration_time; i++) {
-        for(uint8_t j = 0; j < 6; j++) {
-            sum[j] += data[j][i];
+
+   for(int a=0;a<11;a++){                                        // I decide that better option is to calibrate accel and gyro separately
```

Comment:
Czemu jest to lepsze? Kalibracja wtedy jest 2x dłuższa

Reply:
Ale osobno robi gyro i Accel jedno na drugie nie wpływa